pkg/apis/machine/types.go

```diff
@@ -92,6 +95,27 @@ type MachineTemplateSpec struct {
 	Spec MachineSpec
 }
 
+// MachineControllerConfig describes the configurations useful for the machine-controller.
+type MachineControllerConfig struct {
+	// MachineDrainTimeout is the time out after which machine is deleted force-fully.
+	MachineDrainTimeout *metav1.Duration
+
+	// MachineHealthTimeout is the timeout after which machine is declared unhealthy/failed.
+	MachineHealthTimeout *metav1.Duration
+
+	// MachineCreationTimeout is the timeout after which machinie creation is declared failed.
+	MachineCreationTimeout *metav1.Duration
+
+	// MaxEvictRetries is the number of retries that will be attempted while draining the node.
+	MaxEvictRetries *int32
+
+	// MachineSafetyOrphanVMsPeriod is the time-period after which safety-controller attempts to find the orphan VMs.s
+	MachineSafetyOrphanVMsPeriod *metav1.Duration
```

Comment:
This field wouldn't make sense here I think. Because this is purely related to the machine class and not a machine-specific field.

Reply:
Sure, I shall remove it.

Reply:
done.